pkg/backend/filestate/meta_test.go

```diff
@@ -60,7 +99,11 @@ func TestEnsurePulumiMeta(t *testing.T) {
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.desc, func(t *testing.T) {
-			t.Parallel()
+			if len(tt.env) > 0 {
+				for k, v := range tt.env {
+					t.Setenv(k, v)
+				}
+			}
```

Comment:
I'm just curious, is the `len` check necessary? Why not just:

```suggestion
			for k, v := range tt.env {
				t.Setenv(k, v)
			}
```

Reply:
Length check is not necessary; I realized this while going over the code with @dixler.
This is completely deleted in #12497, but I'll update it here.